deps/cares/include/ares_build.h

```diff
@@ -0,0 +1,117 @@
+/* ares_build.h.  Generated from ares_build.h.in by configure.  */
```

Comment:
It doesn't look like `ares_rules.h` can be moved much. It is included from `ares.h`. I'm not sure about moving build stuff to GYP too. Current directory structure seems to be easier to update.